src/cache.cpp

```diff
@@ -34,23 +34,47 @@
 
 namespace {
 	typedef std::pair<std::string,std::string> string_pair;
+	typedef std::pair<BitmapRef, uint32_t> bitmap_pair;
```

Comment:
Wouldn't it be better to define a struct here? The `.second.second` stuff is kind of hard on the eyes.